testsuite/libra-fuzzer/src/fuzz_targets/execute_and_commit_chunk.rs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::FuzzTargetImpl;
+use executor::{
+    fuzzing::{FakeDb, FakeVM},
+    Executor,
+};
+use executor_types::ChunkExecutor;
+use libra_proptest_helpers::ValueGenerator;
+use libra_types::{ledger_info::LedgerInfoWithSignatures, transaction::TransactionListWithProof};
+use proptest::{
+    arbitrary::Arbitrary,
+    strategy::{Strategy, ValueTree},
+    test_runner::{self, TestRunner},
+};
+use rand::RngCore;
+use storage_interface::DbReaderWriter;
+
+#[derive(Clone, Debug, Default)]
+pub struct ExecuteAndCommitChunk;
+
+impl FuzzTargetImpl for ExecuteAndCommitChunk {
+    fn name(&self) -> &'static str {
+        module_name!()
+    }
+
+    fn description(&self) -> &'static str {
+        "state-sync > executor::execute_and_commit_chunk"
+    }
+
+    fn generate(&self, _idx: usize, _gen: &mut ValueGenerator) -> Option<Vec<u8>> {
+        let mut output = vec![0u8; 4096];
+        let mut rng = rand::thread_rng();
+        rng.fill_bytes(&mut output);
+        Some(output)
+    }
+
+    fn fuzz(&self, data: &[u8]) {
+        // setup fake db
+        let fake_db = FakeDb {};
+        let db_reader_writer = DbReaderWriter::new(fake_db);
+        let mut executor = Executor::<FakeVM>::new(db_reader_writer);
+
+        // construct arguments
+        let passthrough_rng =
+            test_runner::TestRng::from_seed(test_runner::RngAlgorithm::PassThrough, &data);
+        let config = test_runner::Config::default();
+        let mut runner = TestRunner::new_with_rng(config, passthrough_rng);
+
+        // txn_list_with_proof
+        let strategy = TransactionListWithProof::arbitrary();
+        let strategy_tree = match strategy.new_tree(&mut runner) {
+            Ok(x) => x,
+            Err(_) => return,
+        };
+        let txn_list_with_proof = strategy_tree.current();
+
+        // verified_target_li
+        let strategy = LedgerInfoWithSignatures::arbitrary();
```

Comment:
how do you make sure the LI is consistent with txn_list?

Reply:
right now it isn't, because it doesn't have to be IIRC (state sync doesn't verify it).

but the fuzzer is probably not going to get passed `verify_chunk()` because of that.

We can fix this to always have a consistent LI, or we can do it in another fuzzer to make sure we cover inputs that don't have consistent LI. What do you think?

Also how do we make it so that the LI is consistent...

Reply:
I just realized it is a draft. I think we verified a lot of stuff before execution the chunk.
`txn_list` should be valid with proof and it should be consistent with the `root_hash` carried by the LI passed in. Basically you have to generate all the txns backed by a LI and break them into smaller chunks with less txns in each chunk.

Reply:
@sausagee How to test this fuzzing function? According to what I said, I don't think it can pass through the verification process.

Reply:
check the README in libra-fuzzer, tl;dr is:

```
cargo run -p libra-fuzzer --bin libra-fuzzer generate execute_and_commit_chunk
cargo run -p libra-fuzzer --bin libra-fuzzer fuzz execute_and_commit_chunk
```

Reply:
state-sync does the following checks ([according to this](https://github.com/libra/libra/blob/master/state-synchronizer/src/coordinator.rs#L925)):

* `target_epoch_and_version = (target.ledger_info().epoch(), target.ledger_info().version())`
* 
   ```
   local_epoch_and_version = (
      self.local_state.highest_local_li.ledger_info().epoch(),
      self.local_state.highest_local_li.ledger_info().version(),
   );
   ```
* ensure `target_epoch_and_version >= local_epoch_and_version`

Reply:
for the first test, the code should just return an error anyway instead of crashing so we should be able to fuzz it without change of the proptest strategy: https://github.com/libra/libra/blob/master/execution/executor/src/lib.rs#L184

Reply:
made a PR for this: https://github.com/libra/libra/pull/5454

Reply:
for the second test we can maybe bootstrap a local epoch = 0 & version = 0